pkg/kubectl/resource/visitor.go

```diff
@@ -441,6 +441,9 @@ func FilterNamespace(info *Info) error {
 // set. If info.Object is set, it will be mutated as well.
 func SetNamespace(namespace string) VisitorFunc {
 	return func(info *Info) error {
+		if !info.Namespaced() {
```

Comment:
No, namespace is always provided into the builder, but this is specifically "set namespace if it can be set", not "hey, I expect set namespace to work on every object I pass it".

----- Original Message -----
> > @@ -441,6 +441,9 @@ func FilterNamespace(info *Info) error {
> >  // set. If info.Object is set, it will be mutated as well.
> >  func SetNamespace(namespace string) VisitorFunc {
> >  	return func(info *Info) error {
> > +		if !info.Namespaced() {
> 
> should this err if `!info.Namespaced() && len(namespace) > 0`?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7969/files#r29959580